src/test/java/com/fasterxml/jackson/databind/ObjectReaderTest.java

```diff
@@ -805,4 +808,111 @@ public int hashCode() {
             return _delegate.hashCode();
         }
     }
+
+
+    public void testReadValueFromFile() throws IOException {
+        File file = _createFileWithNameAndJson(
+            "testReadValueFromFile.json",
+            a2q("{ 'name': 'John Doe'}"));
+
+        FilePerson bean = newJsonMapper().readerFor(FilePerson.class).readValue(file);
+
+        assertEquals("John Doe", bean.name);
+        _cleanUpFromFileSystem(file);
+    }
+
+    public void testReadValueFromFile2() throws IOException {
+        File file = _createFileWithNameAndJson(
+            "testReadValueFromFile2.json",
+            a2q("{ 'name': 'John Doe'}"));
+
+        FilePerson bean = newJsonMapper().reader().readValue(file, FilePerson.class);
+
+        assertEquals("John Doe", bean.name);
+        _cleanUpFromFileSystem(file);
+    }
+
+    public void testReadValueFromFileWithFormatDetection() throws IOException {
+        File file = _createFileWithNameAndJson(
+            "testReadValueFromFileWithFormatDetection.json",
+            a2q("{ 'name': 'John Doe'}"));
+        ObjectReader reader = newJsonMapper().readerFor(FilePerson.class);
+        reader = reader.withFormatDetection(reader);
+
+        FilePerson bean = reader.readValue(file);
+
+        assertEquals("John Doe", bean.name);
+        _cleanUpFromFileSystem(file);
+    }
+
+    public void testReadValueFromNonExistentFile() throws IOException {
+        File file = new File("testReadValueFromNonExistentFile.json");
```

Comment:
Maybe define MISSING_FILE constant, use for all methods, no need to have different missing files.

Reply:
Oh sorry maybe I misread. No, if there's just one in this class that's fine.
No need to try to combine across test files.